OpenLaTeXStudio/Editor/src/main/java/latexstudio/editor/menu/OpenTemplate.java

```diff
@@ -144,6 +144,14 @@ private Template getTemplateFromElement(NodeList templateNodes) {
 
     @Override
     public void actionPerformed(ActionEvent e) {
-        this.setVisible(true);
+        Enum canOpen = etc.canOpen();
+        
+        if (canOpen == EditorTopComponent.CanOpenState.SAVE_AND_OPEN) {
+            etc.fileAction().saveFile();
+            this.setVisible(true);
+            
+        } else if (canOpen == EditorTopComponent.CanOpenState.OPEN) {
+            this.setVisible(true);
```

Comment:
I'd merge default with CANCEL. It's correct that it currently cannot return anything else, but: 1. It's a good practice to have default clause on switches 2. an enum can always be expanded